spec/outputs/syslog_tls_spec.rb

```diff
@@ -0,0 +1,165 @@
+    rescue Errno::EADDRINUSE
+      retry
+    end
+  end
+
+  let(:server) { TCPServer.new("127.0.0.1", port) }
+
+  shared_examples "syslog output" do
+    it "should write expected format" do
+      Thread.start { sleep 0.25; subject.receive event }
+      socket = secure_server.accept
+      read = socket.sysread(100)
+      expect(read.size).to be > 0
+      expect(read).to match(output)
+    end
+  end
+
+  context "connects with TLS" do
+    let(:event) { LogStash::Event.new({ "message" => "foo bar", "host" => "baz" }) }
+    let(:options) { { "host" => "localhost", "port" => port, "protocol" => "ssl-tcp",
+      "ssl_cacert" => File.join(FIXTURES_PATH, "ca.pem"),
+      "ssl_cert" => File.join(FIXTURES_PATH, "client.pem"),
+      "ssl_key" => File.join(FIXTURES_PATH, "client-key.pem") } }
+    # The output details are tested in syslog_spec.rb so simply check for message to be present.
+    let(:output) { /foo bar/ }
+
+    let(:secure_server) do
+      # Create TLS server with given certificate and private key, and verify client certificate against CA.
+      ssl_context = OpenSSL::SSL::SSLContext.new
+      ssl_context.cert = OpenSSL::X509::Certificate.new(File.read(server_cert_file))
+      ssl_context.key = OpenSSL::PKey::read(File.read(server_pkey_file), nil)
+      ssl_context.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      ssl_context.cert_store = OpenSSL::X509::Store.new
+      ssl_context.cert_store.add_cert(OpenSSL::X509::Certificate.new(File.read(File.join(FIXTURES_PATH, "ca.pem"))))
+      OpenSSL::SSL::SSLServer.new(server, ssl_context)
+    end
+
+    after(:each) do
+      secure_server.close rescue nil
+    end
+
+    context "server with valid certificates" do
+      let(:options ) { super().merge("ssl_verify" => true) }
+      let(:server_cert_file) { File.join(FIXTURES_PATH, "valid-server.pem") }
+      let(:server_pkey_file) { File.join(FIXTURES_PATH, "valid-server-key.pem") }
+
+      it_behaves_like "syslog output"
+    end
+
+    context "ssl_verify == false, server with untrusted certificates" do
+      let(:options ) { super().merge("ssl_verify" => false) }
+      let(:server_cert_file) { File.join(FIXTURES_PATH, "untrusted-server.pem") }
+      let(:server_pkey_file) { File.join(FIXTURES_PATH, "untrusted-server-key.pem") }
+
+      it_behaves_like "syslog output"
+    end
+
+    context "ssl_verify == true, server with untrusted certificates" do
+      let(:options ) { super().merge("ssl_verify" => true) }
+      let(:server_cert_file) { File.join(FIXTURES_PATH, "untrusted-server.pem") }
+      let(:server_pkey_file) { File.join(FIXTURES_PATH, "untrusted-server-key.pem") }
+
+      it "syslog output refuses to connect" do
+        Thread.start { secure_server.accept rescue nil }
+        expect(subject.logger).to receive(:error).with(/SSL Error/i, hash_including(exception: OpenSSL::SSL::SSLError)).once.and_throw :TEST_DONE
+        expect { subject.receive event }.to throw_symbol(:TEST_DONE)
+      end
+    end
```

Comment:
```suggestion
    context "server with untrusted certificates" do
      let(:server_cert_file) { File.join(FIXTURES_PATH, "untrusted-server.pem") }
      let(:server_pkey_file) { File.join(FIXTURES_PATH, "untrusted-server-key.pem") }

      context "ss_verify disabled" do
        let(:options ) { super().merge("ssl_verify" => false) }

        it_behaves_like "syslog output"
      end      

      context "ss_verify enabled" do
        let(:options ) { super().merge("ssl_verify" => true) }

        it "should refuse to connect" do
          Thread.start { secure_server.accept rescue nil }
          expect(subject.logger).to receive(:error).with(/SSL Error/i, hash_including(exception: OpenSSL::SSL::SSLError)).once.and_throw :TEST_DONE
          expect { subject.receive event }.to throw_symbol(:TEST_DONE)
        end
      end
    end
```

A nested context could make it even more readable

Reply:
True, changed accordingly!